lib/src/main/java/graphql/nadel/engine/blueprint/NadelFastSchemaTraverser.kt

```diff
@@ -0,0 +1,272 @@
+package graphql.nadel.engine.blueprint
+
+import graphql.nadel.engine.util.unwrapAll
+import graphql.schema.GraphQLAppliedDirective
+import graphql.schema.GraphQLAppliedDirectiveArgument
+import graphql.schema.GraphQLArgument
+import graphql.schema.GraphQLCompositeType
+import graphql.schema.GraphQLDirective
+import graphql.schema.GraphQLDirectiveContainer
+import graphql.schema.GraphQLEnumType
+import graphql.schema.GraphQLEnumValueDefinition
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLFieldsContainer
+import graphql.schema.GraphQLInputFieldsContainer
+import graphql.schema.GraphQLInputObjectField
+import graphql.schema.GraphQLInputObjectType
+import graphql.schema.GraphQLInterfaceType
+import graphql.schema.GraphQLList
+import graphql.schema.GraphQLModifiedType
+import graphql.schema.GraphQLNamedSchemaElement
+import graphql.schema.GraphQLNamedType
+import graphql.schema.GraphQLNonNull
+import graphql.schema.GraphQLNullableType
+import graphql.schema.GraphQLObjectType
+import graphql.schema.GraphQLOutputType
+import graphql.schema.GraphQLScalarType
+import graphql.schema.GraphQLSchema
+import graphql.schema.GraphQLSchemaElement
+import graphql.schema.GraphQLType
+import graphql.schema.GraphQLTypeReference
+import graphql.schema.GraphQLUnionType
+import graphql.schema.GraphQLUnmodifiedType
+
+/**
+ * Significantly faster than normal [graphql.schema.SchemaTraverser] as it's simpler.
+ *
+ * Validating central schema with old [graphql.schema.SchemaTraverser] takes 306ms and this impl takes 74ms.
+ *
+ * That's 4x faster.
```

Comment:
I reverted the changes in f7c095e318d15f39ba1094fdcfad1f697be80313 to show this.

I also reapplied relevant optimisations to the file that weren't related to the `SchemaTraverser` so it should be a pure comparison.